discovery/consul/consul_discovery_test.go

```diff
@@ -36,14 +54,19 @@ func (rt *ConsulChangingRoundTripper) RoundTrip(req *http.Request) (*http.Respon
 	rec := httptest.NewRecorder()
 	if req.URL.Path == "/v1/health/service/forwardServiceName" {
 		var resp []byte
-		if rt.Count == 2 {
-			// On the second invocation, return zero hosts!
-			resp, _ = ioutil.ReadFile("testdata/consul/health_service_zero.json")
-		} else if rt.Count == 1 {
-			// On the second invocation, return two hosts!
-			resp, _ = ioutil.ReadFile("testdata/consul/health_service_two.json")
-		} else {
-			resp, _ = ioutil.ReadFile("testdata/consul/health_service_one.json")
+		var err error
+		switch rt.Count {
+		case 1:
+			// On the first invocation, return two hosts
+			resp, err = ioutil.ReadFile("testdata/health_service_two.json")
+		case 2:
+			// On the second invocation, return zero hosts
+			resp, err = ioutil.ReadFile("testdata/health_service_zero.json")
+		default:
+			resp, err = ioutil.ReadFile("testdata/health_service_one.json")
+		}
+		if err != nil {
+			return nil, err
```

Comment:
From this doc: https://pkg.go.dev/net/http#RoundTripper, I'm interpreting the "if" in

> In particular, RoundTrip must return err == nil if it obtained a response, regardless of the response's HTTP status code.

as an "if and only if", so that we either return `nil, err` or `httpResponse, nil`. I don't think we should be returning an http error code here, since in this case, we are not testing an http error response, but are instead signaling a failure of the test harness (i.e. to load the response that should be sent). Thoughts?